carsus/io/output/tardis_.py

```diff
@@ -505,44 +505,37 @@ def get_energies(method):
 
         return data
 
-    @staticmethod
-    def _get_all_lines_data(lines_q):
-
+    def _get_all_lines_data(self):
+        levels_subq = self._build_levels_q()
 
-        lines = list()
-        for line in yield_limit(lines_q.options(joinedload(Line.wavelengths), joinedload(Line.gf_values)),
-                                Line.line_id, maxrq=LINES_MAXRQ):
-            try:
-                # Try to get the first gf value
-                gf = line.gf_values[0].quantity
-            except IndexError:
-                print "No gf value is available for line {0}".format(line.line_id)
-                continue
-            try:
-                # Try to get the first wavelength
-                wavelength = line.wavelengths[0]
-            except IndexError:
-                print "No wavelength is available for line {0}".format(line.line_id)
-                continue
+        wavelength = aliased(LineWavelength)
+        gf = aliased(LineGFValue)
 
-            if wavelength.medium == MEDIUM_VACUUM:
-                wavelength_value = wavelength.quantity.value
-            elif wavelength.medium == MEDIUM_AIR:
-                wavelength_value = convert_wavelength_air2vacuum(wavelength.quantity.value)
-            else:
-                raise AtomDataUnrecognizedMediumError(
-                    "The medium {} is not recognized (0 - vacuum, 1 - air)".format(wavelength.medium)
+        lines_q = (
+                self.session.
+                query(
+                    Line.line_id,
+                    Line.lower_level_id,
+                    Line.upper_level_id,
+                    wavelength._value.label('wavelength'),
+                    gf._value.label('gf'),
```

Comment:
@mishinma Thanks for pointing this out, I tested your syntax and it works. I should write a short documentation about this nice feature